bindings/js/api.js

```diff
@@ -275,16 +276,13 @@ var HaclWasm = (function() {
       }
       return read_memory(pointer.value, size);
     });
-    if (return_buffers.length == 1) {
```

Comment:
While adding tests for NaCl I had to make one last change to api.json on which I'd like to get @denismerigoux's approval before merging. Here we were flattening the return_buffers array only if it contained exactly one buffer, which is the case for almost all the functions in the API. A function would thus return
1. `[ ret ]` if it has no output buffers
2. `output_buf` if its return type is void and it has 1 output buffer
3. `[ ret, output_buf ]` if its return type is not void and it has 1 output buffer
4. `[ ret, [ output_buf1, output_buf2, ...]]` if its return type is not void and it has more than 1 output buffer

With these changes, case 1 and 3 remain unchanged and case 2 and 4 are now:
2. `[ output_buf ]` if its return type is void and it has 1 output buffer
4. `[ ret, output_buf1, output_buf2, ...]` if its return type is not void and it has more than 1 output buffer

So now, irrespective of return type and number of output values, all JS functions will return an array in which the first element is the return value (absent if return type if void) and the rest of the elements are output buffers.

Reply:
I've made this change because I think it's a bit cleaner and it simplifies the logic required to consume these functions.

Reply:
I'm fine with that, it's indeed more regular